### PR TITLE
Llm1

### PR DESCRIPTION
**Pull request by regislebrun**
- Fix setInitialSearch docstring: LHS -> Sobol' sequence
- Add Python edge case test for discrete distributions
    
    t_MeasureEvaluation_edge.py covers:
    - IndividualChanceMeasure with discrete distribution
    - JointChanceMeasure with discrete distribution
    - QuantileMeasure with discrete distribution
    - MeanStdTradeoff/Variance/Aggregated with discrete distribution
    - Multi-output IndividualChanceMeasure with discrete distribution
    - Save/load roundtrip for measures with discrete distributions

- Add C++ tests for MeasureFunction, MeasureFactory, and edge cases
    
    - t_MeasureFunction_std: test construction, Point/Sample evaluation,
      dimensions, clone
    - t_MeasureFactory_std: test build(), buildCollection(), shared dist
    - t_MeasureEvaluation_edge: test discrete distributions for all measure
      types (exercises IndividualChanceMeasure discrete branch),
      multi-output IndividualChanceMeasure, copy-on-write behavior

- Refactor SequentialMonteCarloRobustAlgorithm with doMultiStart and solver fallback
    
    Extract multi-start into a dedicated doMultiStart method using Sobol'
    sequence instead of inline LHSExperiment. Add try/catch fallback when
    the local solver fails: if initialSearch > 0, retry with a tight Sobol'
    multi-start around currentPoint.
    
    Fix convergence criterion: remove spurious epsilon < getMaximumAbsoluteError()
    condition that could stop early when inner solver tolerance tightened.

- Fix array indexing and serialization bugs
    
    - IndividualChanceMeasure: fix weights[j] -> weights[i] (wrong loop variable
      in discrete branch)
    - QuantileMeasure: fix outS(i,0) -> outS(j,0) (wrong row index in wrapper
      Sample operator)
    - SubsetInverseSamplingResult: add missing threshold_ serialization
    - SubsetInverseSampling: add missing thresholdDistribution_ serialization

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected chance/quantile measure probability handling and output placement for multi-output evaluations.
  * Improved numerical stability by clamping intermediate probabilities before quantile conversion.
* **New Features**
  * Updated robust optimization multi-start initialization to use Sobol low-discrepancy sampling, with improved iteration logging and revised convergence checks.
  * Added a retry strategy for local-optimization failures via localized multistart when enabled.
* **Persistence**
  * Extended inverse-subset sampling results/threshold data to be saved and restored.
* **Tests**
  * Added new edge-case tests for measure evaluation and factory/function wrappers; updated expected outputs.
* **Documentation**
  * Refreshed docstrings/theory text to reflect Sobol-based multi-start behavior.

### CODE REVIEW

File: lib/src/SequentialMonteCarloRobustAlgorithm.cxx [hunk 1]
```diff
@@ -101,17 +119,20 @@ void SequentialMonteCarloRobustAlgorithm::run()
   setResult(OptimizationResult(robustProblem));
 
   UnsignedInteger iterationNumber = 0;
+  LOGINFO("Start main loop");
   while ((!convergence) && (iterationNumber <= getMaximumIterationNumber()))
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/SequentialMonteCarloRobustAlgorithm.cxx:124]: fixed

File: python/test/t_MeasureEvaluation_edge.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+study.add('jcm', jcm)
+study.save()
+
+study2 = ot.Study()
+study2.setStorageManager(ot.XMLStorageManager(filename))
+study2.load()
+icm_loaded = otrobopt.IndividualChanceMeasure()
+study2.fillObject('icm', icm_loaded)
+print('IndividualChance loaded=', icm_loaded)
+os.remove(filename)
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply python/test/t_MeasureEvaluation_edge.py]: fixed

File: lib/src/SequentialMonteCarloRobustAlgorithm.cxx [hunk 3]
```diff
@@ -162,17 +175,41 @@ void SequentialMonteCarloRobustAlgorithm::run()
...
-      newValue = result.getOptimalValue();
-    }
+      try
+      {
+	solver.run();
+	result = solver.getResult();
+	if (result.getOptimalPoint().getDimension() == 0) throw InvalidArgumentException(HERE) << "Local optimization failed, try multistart if allowed";
+      }
+      catch (const InvalidArgumentException & ex)
+      {
```
Comment (jschueller) [inline lib/src/SequentialMonteCarloRobustAlgorithm.cxx:185]: the indentation is messed up

File: python/src/CMakeLists.txt [hunk 4]
```diff
@@ -33,7 +33,10 @@ macro (ot_add_python_module MODULENAME SOURCEFILE)
   target_link_libraries (${module_target} Python::Module)
 
   set_target_properties (${module_target} PROPERTIES OUTPUT_NAME ${MODULENAME})
-  set_target_properties (${module_target} PROPERTIES COMPILE_FLAGS "${SWIG_COMPILE_FLAGS}")
+  if(SWIG_COMPILE_FLAGS)
+    string(REPLACE " " ";" SWIG_COMPILE_FLAGS_LIST "${SWIG_COMPILE_FLAGS}")
+    target_compile_options(${module_target} PRIVATE ${SWIG_COMPILE_FLAGS_LIST})
+  endif()
```
Comment (jschueller) [inline python/src/CMakeLists.txt:39]: I think we should use separate_arguments

File: lib/src/IndividualChanceMeasure.cxx
Comment (jschueller) [inline lib/src/IndividualChanceMeasure.cxx:32]: we should remove the ot<1.27 path

File: lib/src/IndividualChanceMeasure.cxx [hunk 5]
```diff
@@ -179,11 +180,18 @@ Point IndividualChanceMeasure::operator()(const Point & inP) const
...
       } // for j
     } // for i
   } // discrete
-  return operator_.operator()(1.0, 2.0) ? alpha_ - outP : outP - alpha_;
+  Point result(outP.getDimension());
+  const Scalar eps = getPDFThreshold();
+  for (UnsignedInteger i = 0; i < result.getDimension(); ++ i)
+  {
+    const Scalar p = std::min(std::max(outP[i], eps), 1.0 - eps);
+    result[i] = operator_.operator()(1.0, 2.0) ? DistFunc::qNormal(alpha_[i]) - DistFunc::qNormal(p) : DistFunc::qNormal(p) - DistFunc::qNormal(alpha_[i]);
```
Comment (jschueller) [inline lib/src/IndividualChanceMeasure.cxx:192]: why the use of normal quantile ?
I dont see it in the formula

File: lib/src/JointChanceMeasure.cxx [hunk 6]
```diff
@@ -179,7 +182,8 @@ Point JointChanceMeasure::operator()(const Point & inP) const
       if (allOk) outP[0] += weights[i];
     } // for i
   }
-  outP[0] = operator_.operator()(1.0, 2.0) ? alpha_ - outP[0] : outP[0] - alpha_;
+  const Scalar p = std::min(std::max(outP[0], getPDFThreshold()), 1.0 - getPDFThreshold());
+  outP[0] = operator_.operator()(1.0, 2.0) ? DistFunc::qNormal(alpha_) - DistFunc::qNormal(p) : DistFunc::qNormal(p) - DistFunc::qNormal(alpha_);
```
Comment (jschueller) [inline lib/src/JointChanceMeasure.cxx:186]: idem
